src/Http/Routing/src/EndpointGroupNameMetadata.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    /// <summary>
+    /// Specifies an endpoint group name in <see cref="Microsoft.AspNetCore.Http.Endpoint.Metadata"/>.
+    /// </summary>
+    public class EndpointGroupNameMetadata : IEndpointGroupNameMetadata
```

Comment:
Is this type necessary? Why not use `EndpointGroupNameAttribute` in `WithGroupName` method?